### PR TITLE
fixing unicode business with BytesIO and encode

### LINKED ISSUE #2: Doesn't work with Python 3 (Unicode)
When running this against an environment created like:

```
conda create -n anaconda-cloud python=3.5
```

First, I was getting some JSON decode errors. These were fixed by using tornado.escape.json_decode instead of json.loads.

After fixing those, I see this when I click the upload button I get an error like the one below:

``` python
    Traceback (most recent call last):
      File "/home/weg/miniconda3/envs/anaconda-notebook/lib/python3.5/site-packages/tornado/web.py", line 1443, in _execute
        result = method(*self.path_args, **self.path_kwargs)
      File "/home/weg/miniconda3/envs/anaconda-notebook/lib/python3.5/site-packages/tornado/web.py", line 2800, in wrapper
        return method(self, *args, **kwargs)
      File "/home/weg/Documents/work/cio/nb_anacondacloud/nb_anacondacloud/nbextension/handlers.py", line 60, in post
        self.finish(json.dumps(uploader.upload()))
      File "/home/weg/Documents/work/cio/nb_anacondacloud/nb_anacondacloud/nbextension/uploader.py", line 48, in upload
        re.sub('\-ipynb$', '', self.name))
      File "/home/weg/miniconda3/envs/anaconda-notebook/lib/python3.5/site-packages/anaconda_client-1.1.1-py3.5.egg/binstar_client/__init__.py", line 412, in upload
        _hexmd5, b64md5, size = compute_hash(fd, size=size)
      File "/home/weg/miniconda3/envs/anaconda-notebook/lib/python3.5/site-packages/anaconda_client-1.1.1-py3.5.egg/binstar_client/utils/__init__.py", line 259, in compute_hash
        hash_obj.update(s)
    TypeError: Unicode-objects must be encoded before hashing
```

This looks pretty hard to fix, as it appears to be buried deep inside the client library (last line).

### PR DESCRIPTION
**Pull request by bollwyvl**
Fixes #2.

This uses `BytesIO` to avoid the Unicode hashing issue. Also tried using binary tempfiles, etc. but this was a smaller change.

This also has #4 in it, too... shouldn't cause too many issues.

### CODE REVIEW

**Comment by damianavila** [comment]
LGTM too... but let's wait for @malev too before merging...

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@ develop-eggs/
 dist/
 .DS_Store
 *.egg-info/
+.ipynb_checkpoints/
```
Comment (malev) [inline .gitignore]: don't think we need this one

Comment (damianavila) [inline .gitignore]: I guess it is something to easy the development work when trying this from the repo...

File: nb_anacondacloud/nbextension/uploader.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 try:
     from StringIO import StringIO
 except ImportError:
-    from io import StringIO
+    from io import BytesIO as StringIO
```
Comment (damianavila) [inline nb_anacondacloud/nbextension/uploader.py]: Importing BytesIO as StringIO could be confusing, so I encourage you to put a little comment here about why you are doing that (and reference the github issue)

**Comment by damianavila** [comment]
Just a tiny comment @bollwyvl

File: .gitignore (hunk 1 above)
Comment (bollwyvl) [inline .gitignore]: Got rid of it. I'll put it in my global git config :)

File: nb_anacondacloud/nbextension/uploader.py (hunk 2 above)
Comment (bollwyvl) [inline nb_anacondacloud/nbextension/uploader.py]: Ha! I didn't know BytesIO was in 2.7. Have changed to BytesIO everywhere.

**Comment by bollwyvl** [comment]
Replied to the above comments, had to expand outdated diff. thanks for the feedback!

**Comment by malev** [comment]
Looks good! I'm guessing that if you merge this one you won't need to merge #4

**Comment by bollwyvl** [comment]
right. I started rebasing with the integration tests, which I plan to talk
about at the code review...

**Comment by damianavila** [comment]
Merging this one as well... I like the incremental updates solving different issues in different PRs... no need to rebase...

**Pull request merged**